### PR TITLE
*: make go/v3 the default plugin

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- hack/generate/samples: update webhook marker and a few other nits
- internal/cmd/operator-sdk/cli: make go/v3 the default

**Motivation for the change:** now that go/v3 is stable, it can be the default plugin. This is not a breaking change because go/v2 is still supported.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
Waiting on https://github.com/kubernetes-sigs/kubebuilder/pull/1882

/hold

**Comment by estroz** [comment]
Blocked by #4315

**Comment by camilamacedo86** [comment]
HHi @estroz, 

It fails  with the error: 

```
warning: error when running kustomize build: kustomize build config/default failed with error: (exec: "kustomize": executable file not found in $PATH) 
```

So, we might need to check if we have not an issue with the makefile target. 

https://github.com/operator-framework/operator-sdk/pull/4307/checks?check_run_id=1553378087#step:5:427

File: changelog/fragments/go-v3-default.yaml
Comment (camilamacedo86) [inline changelog/fragments/go-v3-default.yaml:13]: IMO we need to add here to update the layout from `layout: go.kubebuilder.io/v2` to `layout: go.kubebuilder.io/v3-alpha`
Then, I think would be better the kb link at the end for users are able to look for further info.

**Review by camilamacedo86 — APPROVED** [review]
Just add a few nits. We might want to wait for v3 stable in the kb as well before gets it merged. 
However, /lgtm

File: changelog/fragments/go-v3-default.yaml
Comment (estroz) [reply changelog/fragments/go-v3-default.yaml:13]: I'd rather defer completely to upstream docs for this migration, since you may have to make some other changes before upgrading your project config file's `layout` key.

**Comment by estroz** [comment]
@camilamacedo86 the problem is `TestContext.CleanupManifests()` uses the global `kustomize` binary while go/v3 installs `kustomize` to `./bin/kustomize`. That function needs to be updated to consider a local `kustomize` path. In the meantime I've updated the cleanup step to use `make undeploy`, and added a comment about future refactoring.

**Pull request merged**
